pkg/v1/remote/transport/ping.go

```diff
@@ -125,7 +125,7 @@ func ping(ctx context.Context, reg name.Registry, t http.RoundTripper) (*pingRes
 			return nil, CheckError(resp, http.StatusOK, http.StatusUnauthorized)
 		}
 	}
-	return nil, errors.New(strings.Join(errs, "; "))
```

Comment:
Could we do something like `fmt.Errorf("%s; %w", strings.Join(errs[:len(errs)], ","), errs[len(errs)]` so that `errors.Is` is respected (by the last error), but without taking a new dependency on `multierr`?

Reply:
Sure

Reply:
took a hybrid approach